Collections/Sergey/Collections/MySimpleCollection.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.ObjectModel;
+using System.Linq;
+using Collections.Sergey.Loger;
+
+namespace Collections.Sergey.Collections
+{
+    class MySimpleCollection<T>: Collection<T>
+    {
+        private static ILoger _loger;
+        public MySimpleCollection(ILoger loger)
+        {
+            _loger = loger;
+        }
+        protected override void ClearItems()
+        {
+            _loger.Log("The collection going to be empty", ArrayToLogString());
+            base.ClearItems();
+            _loger.Log("The collection is empty", ArrayToLogString());
+        }
+
+        private string ArrayToLogString()
```

Comment:
because anyone may wanna use ToString() in some other ways, not only to show array items - here it is used more for logger, then for collection
It may be right just for this task, I will do this now, I just tried to avoid useless overrides - in other cases, I think,  it may be not only useless, but troublemaking.